sugarscape.py

```diff
@@ -893,50 +899,48 @@ def updateRuntimeStats(self):
-        self.runtimeStats["totalWealthLost"] += totalWealthLost
-        self.runtimeStats["totalMetabolismCost"] += totalMetabolismCost
-
-        self.runtimeStats["agentStarvationDeaths"] = agentStarvationDeaths
-        self.runtimeStats["agentDiseaseDeaths"] = agentDiseaseDeaths
-        self.runtimeStats["agentCombatDeaths"] = agentCombatDeaths
-        self.runtimeStats["agentAgingDeaths"] = agentAgingDeaths
-        self.runtimeStats["agentsReplaced"] = self.agentsReplaced
-        self.runtimeStats["agentsBorn"] = self.agentsBorn
-        self.agentsReplaced = 0
-        self.agentsBorn = 0
-        self.runtimeStats["agentWealthCollected"] = agentWealthCollected
-        self.runtimeStats["agentWealthTotal"] = agentWealthTotal
-        self.runtimeStats["agentWealthBurnRate"] = agentWealthBurnRate
-        self.runtimeStats["agentMeanTimeToLive"] = agentMeanTimeToLive
-        self.runtimeStats["environmentWealthCreated"] = environmentWealthCreated
-        self.runtimeStats["environmentWealthTotal"] = environmentWealthTotal
-
-        self.runtimeStats["agentTotalMetabolism"] = agentTotalMetabolism
-        self.runtimeStats["totalSickAgents"] = totalSickAgents
+        runtimeStats = {"agentAgingDeaths": agentAgingDeaths, "agentCombatDeaths": agentCombatDeaths, "agentDiseaseDeaths": agentDiseaseDeaths, "agentMeanTimeToLive": agentMeanTimeToLive,
+                        "agentStarvationDeaths": agentStarvationDeaths, "agentTotalMetabolism": agentTotalMetabolism, "agentWealthBurnRate": agentWealthBurnRate,
+                        "agentWealthCollected": agentWealthCollected, "agentWealthTotal": agentWealthTotal, "maxWealth": maxWealth, "meanAge": meanAge, "meanAgeAtDeath": meanAgeAtDeath,
+                        "meanConflictHappiness": meanConflictHappiness, "meanFamilyHappiness": meanFamilyHappiness, "meanHappiness": meanHappiness, "meanHealthHappiness": meanHealthHappiness,
+                        "meanMetabolism": meanMetabolism, "meanMovement": meanMovement, "meanSocialHappiness": meanSocialHappiness, "meanTradePrice": meanTradePrice, "meanWealth": meanWealth,
+                        "meanWealthHappiness": meanWealthHappiness, "meanVision": meanVision, "minWealth": minWealth, "population": numAgents, "sickAgents": sickAgents, "tradeVolume": tradeVolume
+                        }
+
+        if group == None:
+            self.runtimeStats["agentsBorn"] = self.agentsBorn
+            self.runtimeStats["agentsReplaced"] = self.agentsReplaced
+            self.runtimeStats["environmentWealthCreated"] = environmentWealthCreated
+            self.runtimeStats["environmentWealthTotal"] = environmentWealthTotal
+            self.runtimeStats["giniCoefficient"] = self.updateGiniCoefficient()
+            self.runtimeStats["timestep"] = self.timestep
+            self.runtimeStats["totalMetabolismCost"] += totalMetabolismCost
+            self.runtimeStats["totalWealthLost"] += totalWealthLost
```

Comment:
Could be useful to do a per-group Gini calculation. I'll leave it out of this PR since it's out of scope for the immediate need (and looks like a bit of a gnarly change).

The two running total metrics have been removed. They were never used. Good eye.